### PR TITLE
feat: add completion command

### LINKED ISSUE #342: Move completion files to the right directory when installing tldr
The completion files at [bin/completion](https://github.com/tldr-pages/tldr-node-client/tree/master/bin/completion) for `bash` and `zsh` currently aren't copied anywhere, so they cannot be seen by `bash` or `zsh`. 

For `bash` completion files have to go to `/usr/share/bash-completion/completions/`.

For `zsh` completion files have to be in any directory in `$fpath` (e.g. `/usr/share/zsh/functions/Completion/Zsh/`).

Is there a reason this hasn't been done yet?

### PR DESCRIPTION
**Pull request by Zamoca42**
## Description

I've implemented the auto-completion feature suggested in roadmap issue #342.

A new `completion.js` file was created to generate and install auto-completion scripts for both bash and zsh shells. I've also added a `completion` command to `bin/tldr`.

Images showing the feature in action are attached.

## Details

- Added `completion.js` to handle completion script generation and installation
- Updated `bin/tldr` to include a new `completion` command
- Supports both bash and zsh shells
- Automatically adds completion scripts to the user's shell configuration file

I hope this feature makes using tldr even more convenient.

### zsh

zsh-test

### bash

bash-test

## Checklist

Please review this checklist before submitting a pull request.

- [x] Code compiles correctly
- [x] Created tests, if possible
- [x] All tests passing (`npm run test:all`)
- [x] Extended the README / documentation, if necessary

## Related

Closes #342

### CODE REVIEW

**Comment by Zamoca42** [comment]
I noticed some TODO comments in the code that aren't directly related to the current changes:

```shell
> tldr@3.4.0 lint
> eslint lib test bin/tldr


/home/runner/work/tldr-node-client/tldr-node-client/lib/tldr.js
Warning:   [10](https://github.com/tldr-pages/tldr-node-client/actions/runs/10441848068/job/28913635228?pr=442#step:7:11)7:9  warning  Unexpected 'todo' comment: 'TODO: make search into a class also.'  no-warning-comments

/home/runner/work/tldr-node-client/tldr-node-client/lib/utils.js
Warning:   56:3  warning  Unexpected 'todo' comment: 'TODO: remove this'  no-warning-comments

/home/runner/work/tldr-node-client/tldr-node-client/test/search.spec.js
Warning:   [15](https://github.com/tldr-pages/tldr-node-client/actions/runs/10441848068/job/28913635228?pr=442#step:7:16)4:1  warning  Unexpected 'todo' comment: 'TODO write tests for private functions...'  no-warning-comments

✖ 3 problems (0 errors, 3 warnings)
```

File: bin/tldr [hunk 1]
```diff
@@ -24,7 +25,8 @@ program
   .option('-e, --random-example', 'Show a random example')
   .option('-f, --render [file]', 'Render a specific markdown [file]')
   .option('-m, --markdown', 'Output in markdown format')
-  .option('-p, --platform [type]', `Override the current platform [${platforms.supportedPlatforms.join(', ')}]`);
+  .option('-p, --platform [type]', `Override the current platform [${platforms.supportedPlatforms.join(', ')}]`)
+  .option('completion <shell>', 'Generate and add shell completion script to your shell configuration');
```
Comment (spageektti) [inline bin/tldr:29]:
```suggestion
  .option('-c, --completion [shell]', 'Generate and add shell completion script to your shell configuration');
```
Wouldn't it better to use the same syntax as for other options?

Comment (Zamoca42) [reply bin/tldr:29]: I believe the latter option looks better. I didn't add `--` to the completion command because, although it's not technically correct, I wanted to express it as a subcommand.

I initially used `<shell>` to indicate a required argument, but I realized it becomes meaningless when using `.option()`. It seems better to align with the convention by using `[shell]`.

In the case of the former option, the short `-c` is similar to `.option('-c, --clear-cache', 'Clear the local cache');`, so we might need to reconsider this.

Comment (spageektti) [reply bin/tldr:29]: For me the second option is fine

Comment (Zamoca42) [reply bin/tldr]: Changed to `[shell]` for consistency.

File: README.md [hunk 2]
```diff
@@ -120,55 +120,32 @@ you can set the configuration variable `skipUpdateWhenPageNotFound` to `true` (d
...
+source ~/.bashrc
 ```
 
-### Bash
+This command will generate the appropriate completion script and append it to your shell's configuration file (`.zshrc` or `.bashrc`).
 
-```bash
-ln -s bin/completion/bash/tldr ~/.tldr-completion.bash
-```
+After running the completion installation command, restart your shell or reload your configuration file to enable the autocompletion.
```
Comment (renie) [inline README.md:144]: Is this sentence really necessary, given the commands above already reloads the config file?

**Review by renie — CHANGES_REQUESTED** [review]
Just a small consideration on README.
The code itself LGTM. I will run a couple of tests on this branch just to be sure, and comment here as soon as it is all checked.
Thanks for the contribution, @Zamoca42 ! :1st_place_medal:

**Comment by renie** [comment]
It works like a charm, @Zamoca42 ! Thanks a lot! :fire: 
Just have a look at the change requested, please.

File: README.md (hunk 2 above)
Comment (Zamoca42) [reply README.md]: Thank you for your review! :smile: 

That’s correct. Since `source ~/.bashrc` was already mentioned, I don’t think there’s a need to remind it again.

I have updated the README!

**Review by renie — APPROVED** [review]
All good now!

**Review by spageektti — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]
LGTM, Thanks for your contribution. If possible, can you revert the oh my zsh plugin section beside the new `zsh` docs (the `zplug` one isn't required) since it allows using the commands with shortcuts?

**Comment by Zamoca42** [comment]
@kbdharun I have reverted the oh-my-zsh plugin section in README! 
Let me know if there are any other changes needed. 😄
